pkg/scheduler/core/generic_scheduler.go

```diff
@@ -1094,25 +1101,39 @@ func selectVictimsOnNode(
 	potentialVictims := util.SortableList{CompFunc: util.MoreImportantPod}
 	nodeInfoCopy := nodeInfo.Clone()
 
-	removePod := func(rp *v1.Pod) {
-		nodeInfoCopy.RemovePod(rp)
+	removePod := func(rp *v1.Pod) error {
+		niClone := nodeInfoCopy.Clone()
```

Comment:
The returned obj will be disregarded upon error, so we don't need to clone here. The previous version was good.